#! /usr/bin/env python

PACKAGE="drone_land_ar_pid"



from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("enable",              bool_t, 0,    "",                   False)
gen.add("c_prop_trans_x",              double_t, 0,    "const for P-part of translation x",                   0.15,     0, 2)
gen.add("c_int_trans_x",              double_t, 0,    "const for I-part of translation x",                   0.01,  0   , 2)
gen.add("c_deriv_trans_x",              double_t, 0,    "const for D-part of translation x",                   0.0,     0, 2)

gen.add("c_prop_trans_y",              double_t, 0,    "const for P-part of translation y",                   0.15,     0, 2)
gen.add("c_int_trans_y",              double_t, 0,    "const for I-part of translation y",                   0.01,  0   , 2)
gen.add("c_deriv_trans_y",              double_t, 0,    "const for D-part of translation y",                   0.0,     0, 2)


gen.add("c_prop_yaw",              double_t, 0,    "const for P-part of translation",                   0.15,     0, 2)
gen.add("c_int_yaw",              double_t, 0,    "const for I-part of translation",                   0,     0, 2)
gen.add("c_deriv_yaw",              double_t, 0,    "const for D-part of translation",                   0,     0, 2)


exit(gen.generate(PACKAGE, "drone_land_ar_pid", "PidParameter"))
